lib/simperium/bucket-store.js

```diff
@@ -1,113 +1,84 @@
-var Promise = require('promise');
+class StoreProvider {
+	constructor( config ) {
+		this.setup = new Promise( ( resolve, reject ) => config( resolve, reject ) );
```

Comment:
Looking a this now, would it make more sense to move the construction of the `Promise` external to `StoreProvider` and simply pass in the promise itself? I think this is originally my weird code.